include/itkTorchImageDataManager.hxx

```diff
@@ -40,16 +40,16 @@ void
 TorchImageDataManager< TImage >
 ::Allocate()
 {
-  // Allocate memory for the GPU here.
   try
     {
+    // Allocate memory for the GPU.  If there is no GPU this will fail
     c10::TensorOptions options = torch::TensorOptions().dtype( ImageType::TorchValueType ).layout( torch::kStrided ).device( torch::kCUDA ).requires_grad( false );
     m_GPUTensor = torch::empty( m_Size, options );
     m_IsGPUBufferAllocated = true;
     }
   catch (...)
     {
-    // What if there is no GPU?!!!
+    m_IsGPUBufferAllocated = false;
```

Comment:
If allocation with `torch::kCUDA` fails, can we allocate with `torch::kCPU`?

Reply:
Even when allocation with `torch::kCUDA` succeeds, we have an instance with `torch:kCPU` too.  The latter shares its pixel data with the superclass `Image<,>` so memory is not an issue.  Having the `torch:kCPU` let's us use the libtorch functions for copying the pixel data back and forth between the GPU and the CPU / `Image<,>`.

I think this is the best way forward, but I want to think about it some more and then discuss it with you.